utils/doclint/generateDotnetApi.js

```diff
@@ -0,0 +1,695 @@
+
+const path = require('path');
+const Documentation = require('./documentation');
+const XmlDoc = require('./xmlDocumentation')
+const PROJECT_DIR = path.join(__dirname, '..', '..');
+const fs = require('fs');
+const { parseApi } = require('./api_parser');
+const { Type } = require('./documentation');
+const { args } = require('commander');
+const { EOL } = require('os');
+// const { visitAll } = require('../markdown'); // TODO: consider using this instead of manual parsing
+
+const maxDocumentationColumnWidth = 80;
+
+/** @type {Map<string, Documentation.Type>} */
+const additionalTypes = new Map(); // this will hold types that we discover, because of .NET specifics, like results
+/** @type {Map<string, string[]>} */
+const enumTypes = new Map();
+
+let documentation;
+/** @type {Map<string, string>} */
+let classNameMap;
+
+{
+  const typesDir = process.argv[2] || '../generate_types/csharp/';
+  let checkAndMakeDir = (path) => {
+    if (!fs.existsSync(path))
+      fs.mkdirSync(path)
+  };
+
+  const modelsDir = path.join(typesDir, "models");
+  const enumsDir = path.join(typesDir, "enums");
+
+  checkAndMakeDir(typesDir);
+  checkAndMakeDir(modelsDir);
+  checkAndMakeDir(enumsDir);
+
+  documentation = parseApi(path.join(PROJECT_DIR, 'docs', 'src', 'api'));
+  documentation.filterForLanguage('csharp');
+
+  documentation.setLinkRenderer(item => {
+    if (item.clazz)
+      return `<see cref="${translateMemberName("interface", item.clazz.name, null)}"/>`;
+    else if (item.member)
+      return `<see cref="${translateMemberName("interface", item.member.clazz.name, null)}.${translateMemberName(item.member.kind, item.member.name, item.member)}"/>`;
+    else if (item.option)
+      return `<paramref name="${item.option}"/>`;
+    else if (item.param)
+      return `<paramref name="${item.param}"/>`;
+    else
+      throw 'Unknown link format.';
+  });
+
+  // get the template for a class
+  const template = fs.readFileSync("./templates/interface.cs", 'utf-8')
+    .replace('[PW_TOOL_VERSION]', `${__filename.substring(path.join(__dirname, '..', '..').length).split(path.sep).join(path.posix.sep)}`);
+
+  // we have some "predefined" types, like the mixed state enum, that we can map in advance
+  enumTypes.set("MixedState", ["On", "Off", "Mixed"]);
+
+  // map the name to a C# friendly one (we prepend an I to denote an interface)
+  // let name = translateMemberName('interface', element.name, undefined);
+  classNameMap = new Map(documentation.classesArray.map(x => [x.name, translateMemberName('interface', x.name, null)]));
+
+  // map some types that we know of
+  classNameMap.set('Error', 'Exception');
+  classNameMap.set('TimeoutError', 'TimeoutException');
+  classNameMap.set('EvaluationArgument', 'object');
+  classNameMap.set('boolean', 'bool');
+  classNameMap.set('Serializable', 'T');
+  classNameMap.set('any', 'object');
+  classNameMap.set('Buffer', 'byte[]');
+  classNameMap.set('path', 'string');
+  classNameMap.set('URL', 'Uri');
```

Comment:
This is an interesting debate - in theory, I agree with the recommendation from the SDK team, that we should use idiomatic concepts from languages as much as possible. So I wanted to do the same, and we can introduce separate overloads later, that are *not* auto-generated, that simplify the usage.

Reply:
I think this compares to the Path/FileInfo debate.
Using Uri and FileInfo will make the user write more code (`new Uri(myUrlString)` and `new FileInfo(myFilePath)`), and maybe requiring the user to add a new using. On the other side we would do a ToString and pass it to the diver. 
The user losses we lose.

Reply:
@kblok well, the PR I had open to change `Path` into `FileInfo` was closed :) so, right now, that resolves to `string`.  Despite the advice, it might be beneficial for us to use `string`...  As @yury-s suggested, it results in easer to use API.